File: materialm-angular-free-main/main/src/app/Models/Game.ts
export class Game{
    id!: number;
    name!:string;
    startDate!:Date;
    endDate!:Date;
    virtualStartDate!:Date;
    startingamount!:number;
    uniqueCode!:string;
    simulationDays!:number;
    candlesPerDay!:number;
    allowstoploss!:boolean;
    allowtakeprofit!:boolean;
    allowshortpos!:boolean;
    allowlongpos!:boolean;
}

File: Pi5/src/main/java/org/example/pi5/entities/Game.java
package org.example.pi5.entities;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Game {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String name;

    private LocalDateTime startDate;

    private LocalDateTime endDate;
    private boolean Allowstoploss;
    private boolean Allowshortpos;
    private boolean Allowlongpos;
    private boolean Allowtakeprofit;
    private LocalDateTime virtualStartDate;
    @Column(unique = true, nullable = false, updatable = false)
    private String uniqueCode;

    private int simulationDays;
    private int candlesPerDay;

    private double startingAmount;

    public Game(int gameId) {
        this.id=gameId;
    }
    /**/
    @PrePersist
    private void generateUniqueCode() {
        // Generate a unique code using UUID and a prefix, can be customized
        this.uniqueCode = "G-" + UUID.randomUUID().toString().substring(0, 8).toUpperCase();
    }
}
